fe-library-app/src/components/BookDetail/BookDetail.tsx

```diff
@@ -0,0 +1,135 @@
+import { useCallback, useEffect, useState } from 'react'
+
+import { useNavigate, useParams } from 'react-router-dom'
+
+import CreateUpdateBookModal from '../../modals/CreateUpdateBookModal'
+import { BookByIdItemResponse, deleteBook, getBookById } from '../../services/BookService'
+import DefaultBookCover from '../BookList/BookCard/DefaultBookCover.png'
+import style from './BookDetail.module.css'
+
+const BookDetail = () => {
+  const { id } = useParams()
+  const [ book, setBook ] = useState<BookByIdItemResponse>()
+  const navigate = useNavigate()
+  const [ showUpdateBookModal, setShowUpdateBookModal ] = useState(false)
+
+  const getBookInfo = useCallback(() => {
+    getBookById(id as string)
+      .then((response) => {
+        setBook(response.data)
+      })
+      .catch(() => {
+        alert('Error with getting book details')
+      })
+  }, [ id ])
+
+  useEffect(() => {
+    getBookInfo()
+  }, [ getBookInfo ])
+
+  const handleModifyBookClick = () => {
+    navigate('/add_modify/' + (id as string))
```

Comment:
Template string would be more readable here

Reply:
Done.